lib/activity/pull-requests.js

```diff
@@ -59,53 +59,89 @@ function storePRMapping(context) {
   cache.set(cacheKey, context.payload.pull_request.id);
 }
 
-async function onStatus(context, subscription, slack) {
-  // we should obtain the new PR data, not the old one
-  const id = await cache.get(`repo#${context.payload.repository.id}:status#${context.payload.sha}`);
+async function getMessageFromCache(id, subscription) {
   const cacheKey = subscription.cacheKey(`pull#${id}`);
-  const message = await cache.get(cacheKey);
+  return cache.get(cacheKey);
+}
 
-  if (!message) {
-    context.log.debug({ sha: context.payload.sha }, 'Could not find pull request for status');
-    return;
-  }
+async function getCombinedStatusForRef(context, sha) {
+  return (await context.github.repos.getCombinedStatusForRef(context.repo({
+    ref: sha,
+  }))).data.statuses;
+}
+
+async function getChecksListForRef(context, sha) {
+  return (await context.github.checks.listForRef(context.repo({
+    ref: sha,
+  }))).data.check_runs;
+}
+
+async function getPullRequest(context, message) {
+  return (await context.github.pullRequests.get({ ...message.context })).data;
```

Comment:
`message.context` has the **GitHub** owner, repo, and number, but I'm happy to pull it from elsewhere if you think it's better!

Reply:
Yeah I think that would be my preference, so that for future readers of this code `message` continues to refer to the cached Slack **message**, as opposed to other data that is also cached with it.

Mostly a variable naming/clarity nitpick 😅

Reply:
Hmm I do like clarity, but aren't we specifically caching these values so we can easily retrieve a pull request like this later? I'm having trouble finding where to get `pull_number` if not from the cache, and `{ ...message.context }` becomes a little uglier with something like

```
{
    owner: context.payload.repository.owner.login,
    repo: context.payload.repository.name,
    pull_number: ?
}
```

Reply:
My bad! I think i've got something backwards. Looks like this is not actually a new change in this PR and we were always storing PR metadata alongside the Slack mesage?

I think I was imagining that we could get the pull request directly by ID, which is cached separately: https://github.com/integrations/slack/blob/520dac1b47611c1276e26fedf37a2e466d42df7b/lib/activity/pull-requests.js#L64

But that improvement can definitely happen outside of this PR! Sorry 😓

Reply:
No worries! I'll keep it focused and unchanged for now by using `message.context`.